[feature] [null support # 6] Use default null support for array transform functions

61yao opened the pull request:
This PR adds default null support for array transform function:
i.e. if any of the argument is null, the function will return null.

Note that we are not able to fully support null in array related functions because we only store one null bit for the whole multi value row.

See the github issue:
https://github.com/apache/pinot/issues/10252

Design doc:
https://docs.google.com/document/d/1w2eBuZrS73xE2JIw8MYlySy5pNaX9iE8kZr_YAL20CM/edit

snleee commented:
@61yao Can you add a bit more information on the PR message on the change for this PR?

```
We are not able to fully support null in array related functions because we only store one null bit for the whole multi value row.
```

The above message is calling out the current limitation of PR or is this PR addressing the above issue?

61yao commented:
> @61yao Can you add a bit more information on the PR message on the change for this PR?
> 
> ```
> We are not able to fully support null in array related functions because we only store one null bit for the whole multi value row.
> ```
> 
> The above message is calling out the current limitation of PR or is this PR addressing the above issue?

@snleee Sorry for the short description. Added extra explanation. Let me know if you need more context.

snleee commented:
> > @61yao Can you add a bit more information on the PR message on the change for this PR?
> > ```
> > We are not able to fully support null in array related functions because we only store one null bit for the whole multi value row.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The above message is calling out the current limitation of PR or is this PR addressing the above issue?
> 
> @snleee Sorry for the short description. Added extra explanation. Let me know if you need more context.

Can you also link to either design document or gitbhub issue related to the entire null support changes? Any person who review this PR would be interested in knowing the entire picture that we want to implement since each PR will include the partial changes.

KKcorps commented on pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/ArrayBaseTransformFunctionTest.java:
```diff
@@ -76,6 +78,101 @@ public void testArrayTransformFunction() {
+
+    switch (getResultDataType(FieldSpec.DataType.INT)) {
+      case INT:
+        Pair<int[], RoaringBitmap> intResults = transformFunction.transformToIntValuesSVWithNull(_projectionBlock);
```
Why is `SVWithNull` being called here? Shouldn't it be`MVWithNull`

snleee replied:
I think that `SV` here represents the output. For instance, the output of `ArrayAvg` is single value.

snleee reviewed (approved):
LGTM

61yao commented:
> > > @61yao Can you add a bit more information on the PR message on the change for this PR?
> > > ```
> > > We are not able to fully support null in array related functions because we only store one null bit for the whole multi value row.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The above message is calling out the current limitation of PR or is this PR addressing the above issue?
> > 
> > 
> > @snleee Sorry for the short description. Added extra explanation. Let me know if you need more context.
> 
> Can you also link to either design document or gitbhub issue for all the null support related PRs? Any person who review this PR would be interested in knowing the entire picture that we want to implement since each PR will include the partial changes.

yeah. good point. Added the github issue and design doc. @snleee Thanks for reviewing!
